### ISSUE TITLE
Leave group button does not work

### ISSUE THREAD

**Issue by cstiens** [zom/Zom-iOS#394]
Group chat 'leave' button does not work

**Comment by N-Pex** [comment]
@cstiens I guess we need to figure out what the "leave" button should actually do...

**Comment by cstiens** [comment]
@N-Pex Yes. We can discuss this on the scrum a bit more today. But, here are my thoughts about it:

In the social sense, if you leave a group, you are choosing not to be part of it anymore. The group should be removed from your Zom and you should no longer be listed as a group member in the group info for anyone else that's still in group.

On Android, we currently have the ability to 'end chat'. When you do this, the group is removed from your conversations view. When any person in the group chooses to do this they essentially leaving the group. So, that's the functionality we should hook up. There are some confusing UX issues with this now. They are filed in #393.

On iOS, you can swipe to 'delete' a group. When you do this, you are also essentially leaving the group. The group is removed from your conversations view. It is never listed in your friends list.

**Comment by N-Pex** [comment]
Should be fixes in 92.

**Comment by N-Pex** [comment]
@chrisballinger @n8fr8 We talked on the scrum that there was no way to distinguish between "user logged out" and "user left the group" because of the presence stanza "unavailable" in both cases, but having read the MUC spec more I found:

'
Presence stanzas of type "unavailable" reflected by the room MUST contain extended presence information about roles and affiliations; in particular, the 'role' attribute MUST be set to a value of "none" to denote that the individual is no longer an occupant.
'

https://xmpp.org/extensions/xep-0045.html#exit , see right below Example 82.

I implemented this on iOS so that we recognize this case and delete the corresponding occupant from the db in we get a presence.type="unavailable" and item.role="none", seems to work well! =)
